MapboxNavigation/NavigationViewController.swift

```diff
@@ -525,6 +535,14 @@ extension NavigationViewController: NavigationServiceDelegate {
         delegate?.navigationViewController(self, didFailToRerouteWith: error)
     }
     
+    public func navigationService(_ service: NavigationService, didRefresh routeProgress: RouteProgress) {
```

Comment:
I think in this place when should redraw all congestion segments (if they were updated) by calling `mapView?.updateRoute(routeProgress)`. At the same time current implementation of congestion segments calculation re-uses already existing `lineGradient` stops which means that route in most recent `RouteProgress` will not be used. Previously I was aiming to prevent re-calculation because for routes with many congestion segments such logic might cause performance drawbacks, on the other hand in current situation there seems no other way.

Reply:
Do you mean in RouteMapViewController?

https://github.com/mapbox/mapbox-navigation-ios/blob/a303150d3a3762799048914d58de793b6ffbd9f4/MapboxNavigation/RouteMapViewController.swift#L522-L524

Reply:
I think most recent update in `RouteMapViewController` should work (removing all route layers, showing them for traveled distance = 0.0, updating them for traveled distance from `RouteProgress`). On the other hand it defats the idea of reusing layers (I think the only data we need from `RouteProgress` in such case is `shape`). We can leave current implementation, but revise it in future.

Reply:
Yeah, let’s deal with any performance optimizations as tail work.